packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx

```diff
@@ -13,31 +13,99 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import React, {
+  HTMLProps,
+  FC,
+  ReactNode,
+  MouseEvent,
+  KeyboardEvent,
+} from 'react';
 import { css } from '@emotion/core';
 import isPropValid from '@emotion/is-prop-valid';
 
 import { focusOutline } from '../../../../styles/style-mixins';
-import { childrenPropType } from '../../../../util/shared-prop-types';
 import { isFunction } from '../../../../util/type-check';
-import { ASCENDING, DESCENDING } from '../../constants';
 import SortArrow from '../SortArrow';
+import styled, { StyleProps } from '../../../../styles/styled';
+import { Direction } from '../../types';
 
-const getAriaSort = (sortable, sortDirection) =>
-  sortable ? sortDirection || 'none' : null;
+export interface TableHeaderProps
+  extends HTMLProps<HTMLTableHeaderCellElement> {
+  /**
+   * Aligns the content of the Header with text-align
+   */
+  align?: 'left' | 'right' | 'center';
+  /**
+   * @private Adds row or col styles based on the provided Scope.
+   * Handled internally
+   */
+  scope?: 'col' | 'row';
+  /**
+   * @private Adds sticky style to the Header based on rowHeader definition.
+   * Handled internally
+   */
+  fixed?: boolean;
+  /**
+   * @private Adds condensed style to the Header based on the table props.
+   * Handled internally
+   */
+  condensed?: boolean;
+  /**
+   * Defines whether or not the Header is sortable
+   */
+  sortable?: boolean;
+  /**
+   * Visually hidden label for the sort button for visually impaired users.
+   * When passed as a function, it is called with the sort `{ direction }`.
+   */
+  sortLabel?: string | (({ direction }: { direction?: Direction }) => string);
+  /**
+   * @private Adds active style to the Header if it is currently hovered by
+   * sort.
+   * Handled internally
+   */
+  isHovered?: boolean;
+  sortDirection?: Direction;
+  /**
+   * @private Adds sorted style to the Header if it is currently sorted
+   * Handled internally
+   */
+  isSorted?: boolean;
+  onClick?: (
+    event:
+      | MouseEvent<HTMLTableHeaderCellElement | HTMLButtonElement>
+      | KeyboardEvent<HTMLTableHeaderCellElement | HTMLButtonElement>,
+  ) => void;
+  /**
+   * @private A testid for selecting table cells in tests. Handled internally.
+   */
+  ['data-testid']?: string;
+  children: ReactNode;
+}
+
+const getAriaSort = (sortable: boolean, sortDirection?: Direction) =>
+  sortable ? sortDirection || 'none' : undefined;
 
-const getLabelSort = (sortable, sortDirection, sortLabel) => {
+const getLabelSort = (
+  sortable?: boolean,
+  sortDirection?: Direction,
+  sortLabel?: string | (({ direction }: { direction?: Direction }) => string),
```

Comment:
Is there a way to reuse these types from the `TableHeaderProps`? I tried to use `Pick` just like in styled components but couldn't 🤔

Reply:
`Pick` returns a new interface with only the properties you picked. Here, you don't want an interface though, since the params aren't in an object shape. Instead, you can access each prop type individually:

```suggestion
  sortable?: TableHeaderProps['sortable'],
  sortDirection?: TableHeaderProps['sortDirection'],
  sortLabel?: TableHeaderProps['sortLabel'],
```

Reply:
Done in 30a8578